src/include/hip/rpp_hip_common.hpp

```diff
@@ -883,16 +1002,182 @@ __device__ __forceinline__ void rpp_hip_pack_float24_pln3_and_store24_pkd3(half
+
+__device__ __forceinline__ void rpp_hip_lds_load8(signed char *srcPtr, int srcIdx, uchar *src_lds)
+{
+    rpp_hip_convert8_i8_to_u8(&srcPtr[srcIdx], src_lds);
+}
+
+// F16 lds loads without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_lds_load8(half *srcPtr, int srcIdx, uchar *src_lds)
+{
+    d_float8 src_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr, srcIdx, &src_f8);
+    rpp_hip_lds_load8((float *)&src_f8, 0, src_lds);
+}
+
+// WITH LAYOUT TOGGLE
+
+// U8 lds loads with layout toggle PKD3 to PLN3 (24 U8 pixels)
+
+__device__ __forceinline__ void rpp_hip_lds_load24_pkd3_to_pln3(uchar *srcPtr, int srcIdx, uchar **src_lds)
+{
+    d_uchar24 *srcPtr_uchar24;
+    srcPtr_uchar24 = (d_uchar24 *)&srcPtr[srcIdx];
+
+    d_uchar8 *src_lds_c1, *src_lds_c2, *src_lds_c3;
+    src_lds_c1 = (d_uchar8 *)src_lds[0];
+    src_lds_c2 = (d_uchar8 *)src_lds[1];
+    src_lds_c3 = (d_uchar8 *)src_lds[2];
+
+    src_lds_c1->x.x = srcPtr_uchar24->x.x.x;
+    src_lds_c1->x.y = srcPtr_uchar24->x.x.w;
+    src_lds_c1->x.z = srcPtr_uchar24->x.y.z;
+    src_lds_c1->x.w = srcPtr_uchar24->y.x.y;
+    src_lds_c1->y.x = srcPtr_uchar24->y.y.x;
+    src_lds_c1->y.y = srcPtr_uchar24->y.y.w;
+    src_lds_c1->y.z = srcPtr_uchar24->z.x.z;
+    src_lds_c1->y.w = srcPtr_uchar24->z.y.y;
+
+    src_lds_c2->x.x = srcPtr_uchar24->x.x.y;
+    src_lds_c2->x.y = srcPtr_uchar24->x.y.x;
+    src_lds_c2->x.z = srcPtr_uchar24->x.y.w;
+    src_lds_c2->x.w = srcPtr_uchar24->y.x.z;
+    src_lds_c2->y.x = srcPtr_uchar24->y.y.y;
+    src_lds_c2->y.y = srcPtr_uchar24->z.x.x;
+    src_lds_c2->y.z = srcPtr_uchar24->z.x.w;
+    src_lds_c2->y.w = srcPtr_uchar24->z.y.z;
+
+    src_lds_c3->x.x = srcPtr_uchar24->x.x.z;
+    src_lds_c3->x.y = srcPtr_uchar24->x.y.y;
+    src_lds_c3->x.z = srcPtr_uchar24->y.x.x;
+    src_lds_c3->x.w = srcPtr_uchar24->y.x.w;
+    src_lds_c3->y.x = srcPtr_uchar24->y.y.z;
+    src_lds_c3->y.y = srcPtr_uchar24->z.x.y;
+    src_lds_c3->y.z = srcPtr_uchar24->z.y.x;
+    src_lds_c3->y.w = srcPtr_uchar24->z.y.w;
+}
+
+// F32 lds loads with layout toggle PKD3 to PLN3 (24 F32 pixels)
+
+__device__ __forceinline__ void rpp_hip_lds_load24_pkd3_to_pln3(float *srcPtr, int srcIdx, uchar **src_lds)
+{
+    d_float24 *srcPtr_f24;
+    srcPtr_f24 = (d_float24 *)&srcPtr[srcIdx];
+
+    d_uint6 src_uchar24;
+    src_uchar24.x.x = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->x.x * (float4) 255.0));
+    src_uchar24.x.y = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->x.y * (float4) 255.0));
+    src_uchar24.y.x = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->y.x * (float4) 255.0));
+    src_uchar24.y.y = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->y.y * (float4) 255.0));
+    src_uchar24.z.x = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->z.x * (float4) 255.0));
+    src_uchar24.z.y = rpp_hip_pack(rpp_hip_pixel_check(srcPtr_f24->z.y * (float4) 255.0));
+
+    d_uchar8 *src_lds_c1, *src_lds_c2, *src_lds_c3;
+    src_lds_c1 = (d_uchar8 *)src_lds[0];
```

Comment:
So due to the NCHW<-> NHWC layout swap, the src_lds[0]/[1]/[2] contain 3 different addresses in the lds for the R/G/B channel. src_lds is basically a double pointer ** in the arguments for this func. Since the ultimate locations in lds are in planar format and aren't consecutive, reading all 24 in one shot didn't work.

Reply:
So due to the NCHW<-> NHWC layout swap, the src_lds[0]/[1]/[2] contain 3 different addresses in the lds for the R/G/B channel. src_lds is basically a double pointer ** in the arguments for this func. Since the ultimate locations in lds are in planar format and aren't consecutive, reading all 24 in one shot didn't work.